backend/linux_input.c

```diff
@@ -101,6 +103,94 @@ static void twin_linux_input_events(struct input_event *ev,
     }
 }
 
+static void *twin_linux_udev_thread(void *arg)
+{
+    struct udev *udev;
+    struct udev_monitor *mon;
+    struct udev_device *dev;
+    int fd;
+
+    /* Create udev object */
+    udev = udev_new();
+    if (!udev) {
+        fprintf(stderr, "Failed to create udev object\n");
+        return NULL;
+    }
+
+    /* Create a monitor for kernel events */
+    mon = udev_monitor_new_from_netlink(udev, "udev");
+    if (!mon) {
+        fprintf(stderr, "Failed to create udev monitor\n");
+        udev_unref(udev);
+        return NULL;
+    }
+
+    /* Filter for input subsystem (keyboard/mouse) */
+    udev_monitor_filter_add_match_subsystem_devtype(mon, "input", NULL);
+    udev_monitor_enable_receiving(mon);
+
+    /* File descriptor for the monitor */
+    fd = udev_monitor_get_fd(mon);
+
+    printf("Listening for keyboard and mouse events...\n");
+
+    while (1) {
+        fd_set fds;
+        FD_ZERO(&fds);
+        FD_SET(fd, &fds);
+
+        int ret = select(fd + 1, &fds, NULL, NULL, NULL);
```

Comment:
Use `poll` system for shorter code.
See https://daniel.haxx.se/docs/poll-vs-select.html

Reply:
Done.